pkg/controlsvc/connect_test.go

```diff
@@ -0,0 +1,207 @@
+package controlsvc_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/ansible/receptor/pkg/controlsvc"
+	"github.com/ansible/receptor/pkg/controlsvc/mock_controlsvc"
+	"github.com/ansible/receptor/pkg/logger"
+	"github.com/golang/mock/gomock"
+)
+
+func CheckExpectedError(expectedError bool, errorMessage string, t *testing.T, err error) {
+	if expectedError {
```

Comment:
These 4 if conditions can be brought down to 2. `if expectedError && errorMessage != err.Error()` and then `if !expectedError && err != nil`